src/utils.ts

```diff
@@ -201,6 +201,13 @@ export namespace Utils {
       }
     }
 
+    // verify homeDomain
```

Comment:
💡  I think it could be mildly beneficial to move this check above the for loop above? I think it may be helpful to debugging if the errors generated by this function for the first manage data operation are returned before checks to the other manage data operations.